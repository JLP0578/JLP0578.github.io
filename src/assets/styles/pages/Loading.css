/* @keyframes typing {
  0%    { width: 0; }
  20%   { width: 100%; content: attr(title); }
  100%  { width: 100%; content: attr(title); }
} */

/* @keyframes blink-cursor {
  0%  { border-color: transparent }
  50% { border-color: #00fd55; }
} */


/***********************************/
/*             DEFAULT             */
/*            TELEPHONE            */
/***********************************/

#preloader {
    position: fixed;
    height: 100vh;
    width: 100vw;
    padding: 5px 15px;

    color: var(--loader-color);
    background-color: var(--loader-bg-color);

    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    align-content: center;
    justify-content: center;
    align-items: center;
}

#preloader .loading-container {
    border: 1px solid var(--loader-color);
    width: 335px;
    padding: 5px 15px;
}

#preloader .loading-container .title {
    background-color: var(--loader-bg-color);
    padding: 0px;
    margin-right: 5px;

    display: block;
}

#preloader .loading-container .title::after {
    content: "";
    opacity: 0;
    animation: none;
}

#preloader .loading-container .loading-bar {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    justify-content: center;
}

#crt {
	pointer-events: none;
}
#crt::before,
#crt::after {
  content: " ";
  display: block;
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

#crt::before {
  background: linear-gradient(rgba(18, 16, 16, 0) 50%, rgba(0, 0, 0, 0.25) 50%), linear-gradient(90deg, rgba(255, 0, 0, 0.06), rgba(0, 255, 0, 0.02), rgba(0, 0, 255, 0.06));
  background-size: 100% 4px, 6px 100%;
}

#crt::after {
  background: rgba(18, 16, 16, .3);
  opacity: 0;
  pointer-events: none;
  animation: flicker 0.15s infinite;
}

@keyframes flicker {
  0% {
    opacity: 0.27861;
  }
  5% {
    opacity: 0.34769;
  }
  10% {
    opacity: 0.23604;
  }
  15% {
    opacity: 0.90626;
  }
  20% {
    opacity: 0.18128;
  }
  25% {
    opacity: 0.83891;
  }
  30% {
    opacity: 0.65583;
  }
  35% {
    opacity: 0.67807;
  }
  40% {
    opacity: 0.26559;
  }
  45% {
    opacity: 0.84693;
  }
  50% {
    opacity: 0.96019;
  }
  55% {
    opacity: 0.08594;
  }
  60% {
    opacity: 0.20313;
  }
  65% {
    opacity: 0.71988;
  }
  70% {
    opacity: 0.53455;
  }
  75% {
    opacity: 0.37288;
  }
  80% {
    opacity: 0.71428;
  }
  85% {
    opacity: 0.70419;
  }
  90% {
    opacity: 0.7003;
  }
  95% {
    opacity: 0.36108;
  }
  100% {
    opacity: 0.24387;
  }
}

#scan-line {
  float: left;
  width: 200%;
  height: 100px;
  z-index: 15;
  position: fixed;

  bottom: 100%;

  background: linear-gradient(0deg,var(--loader-bg-color) 0,rgba(255,255,255,.4) 10%,var(--loader-bg-color) 100%);
  opacity: .1;
  
  pointer-events: none;
  animation: scan-line 8s linear infinite;
}

@keyframes scan-line {
  0%, 70% { 
    bottom: 100%; 
  }
  100% { 
    bottom: calc(0% - 200px); 
  }
}

/***********************************/
/*              CUSTOM             */
/***********************************/
@media screen and (min-width: 400px) { 
    #preloader .loading-container .title {
        padding: 0px 10px;
    
        display: flex;
        flex-direction: row;
        flex-wrap: nowrap;
        justify-content: flex-start;
        align-items: center;
    }

    #preloader .loading-container .title::after {
        content: var(--trois-point);
        opacity: 1;
        min-width: 1.8em;
        animation: 2s dots infinite;
    
        display: flex;
        flex-direction: row;
        flex-wrap: nowrap;
        justify-content: flex-start;
        align-items: center;
    }
    
    @keyframes dots {
        0% {content: var(--un-point);}
        33% {content: var(--deux-point);}
        66% {content: var(--trois-point);}
        100% {content: var(--un-point);}
    }

    .disabled_loader #preloader .perc::after {
        font-size: 1rem;
    }
}

/***********************************/
/*             TABLETTE            */
/***********************************/
@media screen and (min-width: 768px) {
    #preloader {
        padding: 25px 45px;
    }
    #preloader .loading-container {
        width: 360px;
    }
}