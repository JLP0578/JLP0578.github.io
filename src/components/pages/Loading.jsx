import React from 'react';
import PropTypes from 'prop-types';

require('../../assets/styles/pages/Loading.css');

export default class Loading extends React.Component {
    render() {
        return (
        <>
            <section id="preloader">
                <fieldset className="loading-container">
                    <legend className="title" data-content="Loading resources">Loading resources</legend>
                        <div className="loading-bar">
                            <p className="perc"></p>
                        </div>
                </fieldset>
            </section>
            <div id="crt"></div>
            <div id="scan-line"></div>
        </>
        );
    }
}

Loading.propTypes = {
    isLoaded: PropTypes.bool
};

Loading.defaultProps = {
    isLoaded: false
};