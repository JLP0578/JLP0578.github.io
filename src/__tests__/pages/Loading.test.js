import renderer from 'react-test-renderer';

import Loading from '../../components/pages/Loading';

test('renders correctly', () => {
  const tree = renderer
    .create(<Loading />)
    .toJSON();

  expect(tree).toMatchSnapshot();

});