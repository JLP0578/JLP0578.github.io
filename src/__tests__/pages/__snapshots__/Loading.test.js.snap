// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <section
    id="preloader"
  >
    <fieldset
      className="loading-container"
    >
      <legend
        className="title"
        data-content="Loading resources"
      >
        Loading resources
      </legend>
      <div
        className="loading-bar"
      >
        <p
          className="perc"
        />
      </div>
    </fieldset>
  </section>,
  <div
    id="crt"
  />,
  <div
    id="scan-line"
  />,
]
`;
